be/src/runtime/runtime_state.h

```diff
@@ -305,11 +305,8 @@ class RuntimeState {
         load_params->__set_filtered_rows(num_rows_load_filtered());
         load_params->__set_unselected_rows(num_rows_load_unselected());
         load_params->__set_source_scan_bytes(num_bytes_scan_from_source());
-
         // Update datacache load metrics
-        if (config::datacache_enable) {
-            update_load_datacache_metrics(load_params);
-        }
+        update_load_datacache_metrics(load_params);
```

Comment:
```c++
#ifdef USE_STAROS
    if(config::starlet_use_star_cache || config::datacache_enable) {
        update_load_datacache_metrics(load_params);
    }
#elif
    if(config::datacache_enable) {
        update_load_datacache_metrics(load_params);
    }
#endif


void update_load_datacache_metrics(TReportExecStatusParams* load_params) const {
    if (!_query_options.__isset.catalog) {
        return;
    }
    TLoadDataCacheMetrics metrics{};
    metrics.__set_read_bytes(_num_datacache_read_bytes.load(std::memory_order_relaxed));
    metrics.__set_read_time_ns(_num_datacache_read_time_ns.load(std::memory_order_relaxed));
    metrics.__set_write_bytes(_num_datacache_write_bytes.load(std::memory_order_relaxed));
    metrics.__set_write_time_ns(_num_datacache_write_time_ns.load(std::memory_order_relaxed));
    metrics.__set_count(_num_datacache_count.load(std::memory_order_relaxed));

    if (_query_options.catalog == "default_catalog") {
#ifdef USE_STAROS
        xxx
#endif
    } else {
        const BlockCache* cache = BlockCache::instance();
        TDataCacheMetrics t_metrics{};
        DataCacheUtils::set_metrics_from_thrift(t_metrics, cache->cache_metrics());
        metrics.__set_metrics(t_metrics);

        load_params->__set_load_datacache_metrics(metrics);
    }
}
```

How about this? Otherwise we will always call `update_load_datacache_metrics()`

Reply:
what does it affect? it's only a few set